valkey.conf

```diff
@@ -1841,6 +1841,38 @@ aof-timestamp-enabled no
 #
 # cluster-replica-no-failover no
 
+# The timeout in milliseconds for cluster manual failover. If a manual failover
+# does not complete within the specified time, both the replica and the primary
+# will abort it.
+#
+# A manual failover is a special kind of failover that is usually executed when
+# there are no actual failures, but we wish to swap the current primary with one
+# of its replicas (which is the node we send the CLUSTER FAILOVER command to) in
+# a safe way, without any window for data loss.
+#
+# It works in the following way:
+# 1. The replica tells the primary to stop processing queries from clients.
+# 2. The primary replies to the replica with the current replication offset.
+# 3. The replica waits for the replication offset to match on its side, to make
+#    sure it processed all the data from the primary before it continues.
+# 4. The replica starts a failover, obtains a new configuration epoch from the
+#    majority of the masters, and broadcasts the new configuration.
```

Comment:
masters -> primaries

Should we document the complete procedure here in the config file? It is documented in the docs already. We could refer to the docs. I don't mind that we include this here if we keep it short and relevant for the config.

Reply:
I would vote to keep it simple. valkey.conf is end user facing, so the inner workings aren't that relevant in this file.

Reply:
i kind of like this, the conf itself describe the relevant content, usually people may not jump to the doc web, like viewing the configuration items on a terminal.

But i don't have a strong opinion on this.

Reply:
I think it's a problem to have duplicated text. If we fix some typo in one of them, we still have the other copy somewhere and it can be hard to keep both of them in sync. I have seen it before. For example, until some months ago we had a file called `TLS.md` in this repo and `topics/encryption.md` in the doc repo. These files were almost identical, but some new info was added in one of them and some other info was added in the other one.

The config is just a timeout. The caller doesn't need the full procedure to set the timeout. A short description that the replica and the primary wait for each other for some time is useful, but other than that, refer to the CLUSTER FAILOVER command manual please.

Reply:
Longer term, do we want a single source of truth building out both? We could build some JSON files that document what this configs do, then attach it both to the relevant man pages, website pages, and incorporate it into valkey.conf file.